core-build/gulp-core-build/src/tasks/BuildReceiptTask.ts

```diff
@@ -35,21 +35,25 @@ export class CheckBuildReceiptTask extends GulpTask<IBuildReceiptTask> {
     gulp: typeof Gulp,
     completeCallback: (error?: string | Error) => void
   ): Promise<Object> | NodeJS.ReadWriteStream | void {
-    _getLocalHashes().then(localHashes => {
-      _lastLocalHashes = localHashes;
-      _readPackageHashes(path.join(process.cwd(), this.buildConfig.packageFolder, 'build.json')).then(packageHashes => {
-        if (packageHashes) {
-          if (_areObjectsEqual(localHashes, packageHashes)) {
-            this.buildConfig.isRedundantBuild = true;
-            this.log('Build is redundant. Skipping steps.');
-          } else {
-            _areObjectsEqual(localHashes, packageHashes);
-            this.log('Build has new content, continuing execution.');
-          }
-        }
-        completeCallback();
-      });
-    });
+    _getLocalHashes()
+      .then(localHashes => {
+        _lastLocalHashes = localHashes;
+        _readPackageHashes(path.join(process.cwd(), this.buildConfig.packageFolder, 'build.json'))
+          .then(packageHashes => {
+            if (packageHashes) {
+              if (_areObjectsEqual(localHashes, packageHashes)) {
+                this.buildConfig.isRedundantBuild = true;
+                this.log('Build is redundant. Skipping steps.');
+              } else {
+                _areObjectsEqual(localHashes, packageHashes);
+                this.log('Build has new content, continuing execution.');
+              }
+            }
+            completeCallback();
+          })
+          .catch(console.error);
+      })
+      .catch(console.error);
```

Comment:
@nickpape-msft could you comment on this?

What's the right way to report a GCB error?

Reply:
(It's not acceptable for a tool to print an ugly stack dump when it encounters an error.)

---
In reply to: [239222894](https://github.com/Microsoft/web-build-tools/pull/983#discussion_r239222894) [](ancestors = 239222894)

Reply:
This should be

```
((error) => {
  this.logError(error);
});
```

so we get the right framing.

---
In reply to: [239223094](https://github.com/Microsoft/web-build-tools/pull/983#discussion_r239223094) [](ancestors = 239223094,239222894)